builder/amazon/common/step_run_source_instance.go

```diff
@@ -199,14 +199,34 @@ func (s *StepRunSourceInstance) Run(ctx context.Context, state multistep.StateBa
 		return multistep.ActionHalt
 	}
 
-	r, err := ec2conn.DescribeInstances(describeInstance)
-
+	// there's a race condition that can happen because of AWS's eventual
+	// consistency where even though the wait is complete, the describe call
+	// will fail. Retry a couple of times to try to mitigate that race.
+
+	var r *ec2.DescribeInstancesOutput
+	err = retry.Config{
+		Tries: 11,
+		ShouldRetry: func(error) bool {
```

Comment:
Watch out this will be another error

```suggestion
		ShouldRetry: func(err error) bool {
```